src/renderer/state/upload/logics.ts

```diff
@@ -225,6 +225,10 @@ const initiateUploadLogic = createLogic({
       try {
         await fms.upload(upload);
       } catch (error) {
+        if (error?.response?.status === 503) {
+          window.alert("The upload service is undergoing scheduled maintenance (503). Please try again later.");
```

Comment:
Instead of `window.alert()`, we should probably instead wrap this with `dispatch(uploadFailed())` like we see just below this. i haven't acutally checked, but I'm assuming that's the mechanism through which errors are displayed via the stylized error message pop-up rather than the alert dialogue.

Reply:
If this is not straightforward to get working, id be good with a release that uses window.alert, followed up by an improved/styled dialog.